pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```diff
@@ -235,15 +242,15 @@ let
       '';
     }
     # each output is just a symlink to the corresponding container
-    # if the container is missing (that is, outputs == [ ]), create a file, to prevent passing the package to .withPackages
+    # if there are no containers (that is, outputs == [ ]), create a file, to forbid passing the package to .withPackages
     ''
-      for outputName in ''${!outputs[@]} ; do
-        if [[ -n ''${outputDrvs[$outputName]} ]] ; then
+      if [[ ''${#outputDrvs[@]} -gt 0 ]] ; then
+        for outputName in ''${!outputs[@]} ; do
           ln -s "''${outputDrvs[$outputName]}" "''${outputs[$outputName]}"
-        else
-          touch "''${outputs[$outputName]}"
-        fi
-      done
+        done
+      else
+        touch "$out"
+      fi
     '';
 in
-if outputs == [ ] then removeAttrs fakeTeX [ "outputSpecified" ] else build // outputDrvs
+mainDrv
```

Comment:
Please, help me understand this. Why is it safe to replace `build` with `containers.out`?

Reply:
It's more like, `texlive.pkgs` should never export `.build` directly. Otherwise any call to `.withPackages` will depend on all tex, texdoc, etc containers of a package, which would blow up the closure size. `.build` is merely a helper exported under `texlivePackages` to enforce that all containers are built and cached.